feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -0,0 +1,882 @@
+
+			lacpOutPkts := counters.GetLacpOutPkts()
+			if lacpOutPkts == 0 {
+				t.Errorf("counters.GetLacpOutPkts() for %q: got 0, want >0", memberVal.GetInterface())
+			}
+			t.Logf("counters.GetLacpOutPkts() for %q: %d", memberVal.GetInterface(), lacpOutPkts)
+
+			// Check LACP interface status.
+			if !memberVal.GetAggregatable() {
+				t.Errorf("memberVal.GetAggregatable() for %q: got false, want true", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetAggregatable() for %q: %v", memberVal.GetInterface(), memberVal.GetAggregatable())
+
+			if !memberVal.GetCollecting() {
+				t.Errorf("memberVal.GetCollecting() for %q: got false, want true", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetCollecting() for %q: %v", memberVal.GetInterface(), memberVal.GetAggregatable())
+
+			if !memberVal.GetDistributing() {
+				t.Errorf("memberVal.GetDistributing() for %q: got false, want true", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetDistributing() for %q: %v", memberVal.GetInterface(), memberVal.GetAggregatable())
+
+			// Check LCP partner info.
+			if memberVal.GetPartnerId() == "" {
+				t.Errorf("memberVal.GetPartnerId() for %q: got empty string, want non-empty string", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetPartnerId() for %q: %s", memberVal.GetInterface(), memberVal.GetPartnerId())
+
+			if memberVal.GetPartnerKey() == 0 {
+				t.Errorf("memberVal.GetPartnerKey() for %q: got 0, want > 0", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetPartnerKey() for %q: %d", memberVal.GetInterface(), memberVal.GetPartnerKey())
+
+			if memberVal.GetPartnerPortNum() == 0 {
+				t.Errorf("memberVal.GetPartnerPortNum() for %q: got 0, want > 0", memberVal.GetInterface())
+			}
+			t.Logf("memberVal.GetPartnerPortNum() for %q: %d", memberVal.GetInterface(), memberVal.GetPartnerPortNum())
+		}
+	}
+}
+
+func TestP4rtInterfaceID(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp := dut.Port(t, "port1")
+	d := &telemetry.Device{}
+	i := d.GetOrCreateInterface(dp.Name())
+
+	cases := []struct {
+		desc   string
+		portID uint32
+	}{{
+		desc:   "MinPortID",
+		portID: 1,
+	}, {
+		desc:   "AvePortID",
+		portID: math.MaxUint32 / 2,
+	}, {
+		desc:   "MaxPortID",
+		portID: math.MaxUint32,
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			i.Id = ygot.Uint32(tc.portID)
+			dut.Config().Interface(dp.Name()).Replace(t, i)
+
+			// Check path /interfaces/interface/state/id.
+			intfID := dut.Telemetry().Interface(dp.Name()).Id().Lookup(t)
+			if !intfID.IsPresent() {
+				t.Fatalf("intfID.IsPresent() for %q: got false, want true", dp.Name())
+			}
+			t.Logf("Telemetry path/value: %v=>%v:", intfID.GetPath().String(), intfID.Val(t))
+			if intfID.Val(t) != tc.portID {
+				t.Fatalf("intfID.Val(t) for %q: got %d, want %d", dp.Name(), intfID.Val(t), tc.portID)
+			}
+		})
+	}
+}
+
+func TestP4rtNodeID(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	componentName := "SwitchChip3/0"
```

Comment:
Cisco device is not using this name for the component. Could you please make this one per vendor option?

Reply:
Done!